pkg/services/ldap/ldap.go

```diff
@@ -148,10 +154,78 @@ func (server *Server) Login(query *models.LoginUserQuery) (
 	return user, nil
 }
 
+// getUsersIteration is a helper function for Users() method.
+// It divides the users by equal parts for the anticipated requests
+func getUsersIteration(logins []string, fn func(int, int) error) error {
+	lenLogins := len(logins)
+	iterations := int(
+		math.Ceil(
+			float64(lenLogins) / float64(UsersMaxRequest),
+		),
+	)
+
+	for i := 1; i < iterations+1; i++ {
+		var previous int
+		current := i * UsersMaxRequest
+
+		if i == 1 {
+			previous = 0
+		} else {
+			previous = UsersMaxRequest * (i - 1)
+		}
+
+		if lenLogins < current {
+			previous = (i - 1) * UsersMaxRequest
+			current = lenLogins
+		}
+
+		err := fn(previous, current)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
 // Users gets LDAP users
 func (server *Server) Users(logins []string) (
 	[]*models.ExternalUserInfo,
 	error,
+) {
+	var users []*ldap.Entry
+	err := getUsersIteration(logins, func(previous, current int) error {
```

Comment:
As a side note, these requests could be done in parallel, hm, maybe in another iteration?

Reply:
Good idea. I think we can start off by running it like this and parallelize it later if needed.